### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: smclient/smclient.go [hunk 1]
```diff
@@ -295,6 +296,12 @@ func (client *SMClient) register(config *config.Config) (err error) {
 		return aoserrors.Wrap(err)
 	}
 
+	if err := client.sendRuntimeInstanceNotifications(launcher.RuntimeStatus{
```
Comment (al1img) [inline smclient/smclient.go:299]: We should send it in any case? Even if it is nil?

File: logging/archivator.go [hunk 2]
```diff
@@ -134,10 +134,10 @@ func (instance *archivator) sendLog(logID string) (err error) {
 		}).Debugf("Push log")
 
 		instance.logChannel <- cloudprotocol.PushLog{
-			LogID:     logID,
+			LogID:      logID,
```
Comment (al1img) [inline logging/archivator.go:137]: What about to squash this commit with same commit before?

File: logging/logging.go [hunk 3]
```diff
@@ -111,23 +111,14 @@ func (instance *Logging) Close() {
 }
 
 // GetInstanceLog returns instance log.
-func (instance *Logging) GetInstanceLog(request cloudprotocol.RequestLog) error {
+func (instance *Logging) GetInstanceLog(request cloudprotocol.RequestLog) (err error) {
```
Comment (al1img) [inline logging/logging.go:114]: 1. commit message: response
2. You can keep unnamed return value

File: logging/logging_test.go [hunk 4]
```diff
@@ -314,7 +314,7 @@ func TestMaxPartCountLog(t *testing.T) {
 	for {
 		select {
 		case result := <-logging.GetLogsDataChannel():
-			if result.ErrorInfo.Message != "" {
+			if result.ErrorInfo != nil && result.ErrorInfo.Message != "" {
```
Comment (al1img) [inline logging/logging_test.go:317]: Actually if there is ErrorInfo then message should present. You can check only result.ErrorInfo != nil.

File: launcher/envvars.go [hunk 5]
```diff
@@ -103,7 +103,7 @@ func setEnvVarsErr(
 
 func (launcher *Launcher) getInstanceEnvVars(instance InstanceInfo) (envVars []string) {
 	for _, envVarInfo := range launcher.currentEnvVars {
-		if (envVarInfo.ServiceID == instance.ServiceID) &&
+		if (*envVarInfo.ServiceID == instance.ServiceID) &&
```
[the remark this answers is not in the document]
Comment (Eabramov84) [reply launcher/envvars.go:106]: fixed

File: iamclient/iamclient.go [hunk 6]
```diff
@@ -159,6 +159,26 @@ func (client *Client) GetSubjects() (subjects []string) {
 	return client.subjects
 }
 
+// GetCertificate gets certificate by issuer.
+func (client *Client) GetCertificate(certType string) (certURL, keyURL string, err error) {
```
[the remark this answers is not in the document]
Comment (Eabramov84) [reply iamclient/iamclient.go:150]: metho GetCertKeyURL was used. 
GetCertKeyURL - removed

File: logging/logging_test.go (hunk 4 above)
Comment (Eabramov84) [reply logging/logging_test.go:317]: done

File: smclient/smclient.go (hunk 1 above)
Comment (Eabramov84) [reply smclient/smclient.go:299]: no. Fixed

**Pull request closed without merging**